src/game/logic/PlayerLogic.java

```diff
@@ -41,4 +41,21 @@ public boolean isPhaseOne(){
     }
 
     public PlayerToken[] getBoardAsPlayerTokens(){ return gameBoard.getBoard(); }
-}
\ No newline at end of file
+
+    public boolean removePiece(int index) {
+        if (activePlayer.getPlayerToken() == gameBoard.getCell(index).getPlayer()) { // i wrote this
+            return false;
+        }
+
+        if gameBoard.removePieceFromCell(int index) {
+            if (activePlayer == Player1) {
+                Player2.decrementPiecesOnBoard();
+            }
+            else {
+                Player1.decrementPiecesOnBoard()
+            }
```

Comment:
While the logic does work, instead of doing a nested if, perhaps it would be better to just do
```java
activePlayer.decrementPiecesOnBoard();
```

Functionally it's the same, but it makes the logic more straight forward